File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/SeatAssigning/Events/OrderSeatAssignmentsCreated.cs
using System;
using System.Collections.Generic;
using ENode.Eventing;

namespace Registration.SeatAssigning
{
    public class OrderSeatAssignmentsCreated : DomainEvent<Guid>
    {
        public Guid OrderId { get; private set; }
        public IEnumerable<SeatAssignment> Assignments { get; private set; }

        public OrderSeatAssignmentsCreated() { }
        public OrderSeatAssignmentsCreated(Guid orderId, IEnumerable<SeatAssignment> assignments)
        {
            OrderId = orderId;
            Assignments = assignments;
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/SeatAssigning/Events/OrderSeatAssignmentsCreated.java
package com.microsoft.conference.registration.domain.SeatAssigning.Events;

import com.microsoft.conference.registration.domain.SeatAssigning.Models.SeatAssignment;
import org.enodeframework.eventing.DomainEvent;

import java.util.List;

public class OrderSeatAssignmentsCreated extends DomainEvent<String> {
    public String OrderId;
    public List<SeatAssignment> Assignments;

    public OrderSeatAssignmentsCreated() {
    }

    public OrderSeatAssignmentsCreated(String orderId, List<SeatAssignment> assignments) {
        OrderId = orderId;
        Assignments = assignments;
    }
}
